### DISCUSSION TITLE
What is the right way to merge a functional group with another atom?

### DISCUSSION THREAD

**Discussion by yulei-chen** [epam/ketcher#2302] [General]
Hi guys 👋, I found a bug about merging a functional group with another atom, but I'm not sure about the expected behavior. Could you guys give me any ideas about that?

**Steps to Reproduce**
1. Draw a chain
2. Draw a functional group (e.g. Bn)
3. Use _Select Tool_ to drag Bn and drop it on any atom of the chain
4. Click _Undo_
5. Use _Select Tool_ to drag Bn and drop it on the same atom
6. Repeat steps 4&5

**Actual behavior**
After step 3, the Bn connects to the chain.
After every undo and drag&drop, the merged structures are different.

**Expected behavior?** (**Here is what I want to discuss**)
After step 3, should the Bn connect to the chain with an extra bond? or do the behaviors depend on the target atom: (the following logic is the same as Template Tool)
  1. if the target atom hasn't connected to any other atoms, then the target atom should be replaced by Bn
  2. if the target atom has connected to another atom, then the target atom should be replaced by Bn
  3. if the target atom has connected to more than one atom, then the Bn connects to the chain with an extra bond

After every undo and drag&drop, the merging logic should be the same as above.

**Screenshots**

 https://user-images.githubusercontent.com/27288153/223301707-a4758235-2832-40bf-8d7c-e8ccd946f912.mp4

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Windows 10
 - Browser [e.g. chrome, safari] Chrome
 - Version [e.g. 22] Version 110.0.5481.178 (Official Build) (64-bit)

**Ketcher version** [e.g. v2.4.2]. 
[Version 2.8.0-rc.5](https://lifescience.opensource.epam.com/ketcher/index.html)
Build at 2023-03-02; 12:56:58

**Additional context**
https://github.com/epam/ketcher/issues/2110

**Comment by yulei-chen** [comment]
@Zhirnoff @Nitvex @KonstantinEpam23 @St-Permiakov Please take a look at this👀, thank you!

**Reply by Zhirnoff** [reply]
Hi @yuleicul !) We really have such a problem, I suggest adding your description to this task https://github.com/epam/ketcher/issues/2268

**Reply by yulei-chen** [reply]
Thank you! Added.
